pageserver/src/tenant.rs

```diff
@@ -3019,6 +3024,12 @@ impl Tenant {
                     backoff::Cancel::new(CancellationToken::new(), || unreachable!()),
                 )
                 .await?;
+
+                if temp_path.exists() {
+                    tokio::fs::remove_file(&temp_path)
+                        .await
+                        .with_context(|| format!("tempfile removal {temp_path}"))?;
+                }
```

Comment:
We could avoid this exists check by just removing the file and ignoring NotFound.